yaml_edit/lib/src/editor.dart

```diff
@@ -613,10 +589,8 @@ class YamlEditor {
   /// the whole tree.
   ///
   /// [SourceSpan]s in this new tree are not guaranteed to be accurate.
-  YamlNode _deepModify(YamlNode tree, Iterable<Object> path,
-      Iterable<Object> subPath, YamlNode expectedNode) {
-    ArgumentError.checkNotNull(path, 'path');
-    ArgumentError.checkNotNull(tree, 'tree');
+  YamlNode? _deepModify(YamlNode tree, Iterable<Object?> path,
```

Comment:
```suggestion
  YamlNode _deepModify(YamlNode tree, Iterable<Object?> path,
```
Can this return `null`, shouldn't it just return a `YamlNode` with `.value == null` ?